_posts/dev/2014-08-02-kura-dio-apis.md

```diff
@@ -128,6 +130,205 @@ button.setInputListener(new PinListener(){
 		}
 });
 
+```
+### Accessing a GPIO Pin with GPIOService for a switch a Led. 
+
+Inside the **jdk.dio.properties** file:
+
+```
+...
+6 = deviceType: gpio.GPIOPin, pinNumber:6, name:GPIO06
+...
+```
+
+Then in the code:
+
+```java
+
+private GPIOService myservice;
+
+KuraGPIOPin pin = this.myservice.getPinByTerminal(6);
+
+// to allow pin configuration use
+//pin = this.myservice.getPinByTerminal(this.options.isConfigPin());
+
+``` 
+In a **component.xml** enter a new service. Select a **Service** and in a **Referenced Service** open **Add** for add  the GPIOService. Select the **GPIOService** and use edit button for set in the bind field to **bindGPIOService** and set the unbind filed to **unbindGPIOService**.
+Created methods bind and unbind in a principale software.
+
+Then in the code:
+
+```java
+protected synchronized void bindGPIOService(final GPIOService gpioService) {
+        this.myservice = gpioService;
+    }
+
+protected synchronized void unbindGPIOService(final GPIOService gpioService) {
+        this.myservice = null;
+    }
+``` 
+
+
+#### Accessing a GPIO Pin by its index
+
+
+```java
+// Accessing the GPIO Pin number 6. The default behaviour is defined in the
+// jdk.dio.properties file
+//
+// i.e.:
+// gpio.GPIOPin = initValue:0, deviceNumber:0, direction:3, mode:-1, trigger:3
+// 6 = deviceType: gpio.GPIOPin, pinNumber:6, name:GPIO06
+
+close(); //Close the pin
+
+    	
+pin = this.myservice.getPinByTerminal(this.options.isConfigPin());
+    	    
+if (pin == null) {
+return;
+}
+
+open(); //Open the pin
+
+if (this.options.isEnableLed()) {
+	setValue(true); //Turns the LED on
+} else {
+	setValue(false); //Turns the LED off
+	}
+}
+
+```
+
+to simplify the code:
```

Comment:
Why we need the previous snippet? Couldn't we provide directly the final code since, functionally, it is the same?